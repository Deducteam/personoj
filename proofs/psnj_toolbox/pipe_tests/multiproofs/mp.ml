let () =
  let src = "mp.log" in
  let bin = Sys.argv.(1) in
  let out_files = [ "lem1.lp"; "lem2.lp" ] in
  Testing.run_pipe ~src ~out_files bin
