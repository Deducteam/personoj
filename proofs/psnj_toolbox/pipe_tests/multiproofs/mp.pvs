mp : THEORY
BEGIN
  P, Q: bool
  lem1 : LEMMA P IMPLIES Q IMPLIES P
  lem2 : LEMMA P IMPLIES Q IMPLIES Q
END mp
