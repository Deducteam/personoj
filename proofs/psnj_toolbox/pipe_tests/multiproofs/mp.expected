=> lem1.lp:
require open qfo.encoding.lhol qfo.encoding.propositional_connectives;
require qfo.spec.main;
symbol lem1!1 : @Prf (@imp (disj (not qfo.spec.main.P) (disj (not qfo.spec.main.Q) qfo.spec.main.P)) (imp qfo.spec.main.P (imp qfo.spec.main.Q qfo.spec.main.P))) ≔
begin
  why3;
end;
symbol lem1!2 : @Prf (@imp (disj (not qfo.spec.main.P) (disj (not qfo.spec.main.Q) qfo.spec.main.P)) (imp qfo.spec.main.P (imp qfo.spec.main.Q qfo.spec.main.P))) ≔
begin
  why3;
end;
symbol lem1!3 : @Prf (disj (not qfo.spec.main.P) (disj (not qfo.spec.main.Q) qfo.spec.main.P)) ≔
begin
  why3;
end;
symbol lem1!4 : @Prf (@imp (disj (not qfo.spec.main.P) (disj (not qfo.spec.main.Q) qfo.spec.main.P)) (imp qfo.spec.main.P (imp qfo.spec.main.Q qfo.spec.main.P))) ≔
begin
  why3;
end;
=> checked
=> lem2.lp:
require open qfo.encoding.lhol qfo.encoding.propositional_connectives;
require qfo.spec.main;
symbol lem2!5 : @Prf (@imp (disj (not qfo.spec.main.P) (disj (not qfo.spec.main.Q) qfo.spec.main.Q)) (imp qfo.spec.main.P (imp qfo.spec.main.Q qfo.spec.main.Q))) ≔
begin
  why3;
end;
symbol lem2!6 : @Prf (@imp (disj (not qfo.spec.main.P) (disj (not qfo.spec.main.Q) qfo.spec.main.Q)) (imp qfo.spec.main.P (imp qfo.spec.main.Q qfo.spec.main.Q))) ≔
begin
  why3;
end;
symbol lem2!7 : @Prf (disj (not qfo.spec.main.P) (disj (not qfo.spec.main.Q) qfo.spec.main.Q)) ≔
begin
  why3;
end;
symbol lem2!8 : @Prf (@imp (disj (not qfo.spec.main.P) (disj (not qfo.spec.main.Q) qfo.spec.main.Q)) (imp qfo.spec.main.P (imp qfo.spec.main.Q qfo.spec.main.Q))) ≔
begin
  why3;
end;
=> checked
